src/main/java/org/openstreetmap/josm/plugins/pt_assistant/data/PTSegmentToExtract.java

```diff
@@ -0,0 +1,169 @@
+// License: GPL. For details, see LICENSE file.
+package org.openstreetmap.josm.plugins.pt_assistant.data;
+
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.RelationMember;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.sort;
+
+/**
+ * Represents a piece of a route that includes all the ways
+ * belonging to the same bundle of itineraries of vehicles
+ * traveling in the same direction
+ *
+ * @author Polyglot
+ *
+ */
+public class PTSegmentToExtract {
+    private final Relation relation;
+
+    private final ArrayList<RelationMember> ptWays;
+    private List<Integer> indices;
+    private List<String> lineIdentifiers;
+    private List<String> colours;
+    private List<String> streetNames;
```

Comment:
These can all be `private final List`, then in the constructor initialize them all with `new ArrayList<>()` and they can never be `null`.

Edit: `lineIdentifiers` and `colours` could be `Set`s, so they can only contain unique values, using a `TreeSet` it would even be sorted. But as I wrote above, I don't think it's a good idea to add tags with multiple different values for these keys to such segments.